### ISSUE TITLE
[FEATURE] Current gradient accumulation only works for jnp.mean loss

### ISSUE THREAD

**Issue by zhuohan123** [alpa-projects/alpa#268]
The current gradient accumulation only works for jnp.mean loss because we always use mean reduction.
For other losses or auxiliary states, we should support other reduction types such as sum reduction and concatenation reduction.
